File: TeamCode/src/main/java/net/gearmaniacs/teamcode/servos/FoundationServos.java
package net.gearmaniacs.teamcode.servos;

import com.qualcomm.robotcore.hardware.HardwareMap;
import com.qualcomm.robotcore.hardware.Servo;
import net.gearmaniacs.teamcode.utils.IHardware;
import org.jetbrains.annotations.NotNull;

public class FoundationServos implements IHardware {

    private Servo left;
    private Servo right;

    @Override
    public void init(@NotNull HardwareMap hardwareMap) {
        left = hardwareMap.get(Servo.class, "foundation_left");
        right = hardwareMap.get(Servo.class, "foundation_right");
    }

    @NotNull
    public final Servo getLeft() {
        return left;
    }

    @NotNull
    public final Servo getRight() {
        return right;
    }

    public final void attach() {
        left.setPosition(0.0);
        right.setPosition(1.0);
    }

    public final void detach() {
        left.setPosition(1.0);
        right.setPosition(0.0);
    }

    @Override
    public void start() {
    }

    @Override
    public void stop() {
    }
}

File: TeamCode/src/main/java/net/gearmaniacs/teamcode/hardware/servos/FoundationServos.kt
package net.gearmaniacs.teamcode.hardware.servos

import com.qualcomm.robotcore.hardware.HardwareMap
import com.qualcomm.robotcore.hardware.Servo
import net.gearmaniacs.teamcode.utils.IHardware
import net.gearmaniacs.teamcode.utils.extensions.getDevice

class FoundationServos : IHardware {

    lateinit var left: Servo
        private set
    lateinit var right: Servo
        private set

    override fun init(hardwareMap: HardwareMap) {
        left = hardwareMap.getDevice("foundation_left")
        right = hardwareMap.getDevice("foundation_right")
    }

    fun attach() {
        left.position = 0.0
        right.position = 0.8
    }

    fun prepareAttach() {
        left.position = 0.5
        right.position = 0.4
    }

    fun detach() {
        left.position = 1.0
        right.position = 0.0
    }
}
